pkg/certificate/manager.go

```diff
@@ -15,10 +17,13 @@ import (
 	"github.com/openservicemesh/osm/pkg/messaging"
 )
 
+var group singleflight.Group
+
 // NewManager creates a new CertificateManager with the passed MRCClient and options
-func NewManager(ctx context.Context, mrcClient MRCClient, serviceCertValidityDuration time.Duration, msgBroker *messaging.Broker, checkInterval time.Duration) (*Manager, error) {
+func NewManager(ctx context.Context, mrcClient MRCClient, getServiceCertValidityPeriod func() time.Duration, getIngressCertValidityDuration func() time.Duration, msgBroker *messaging.Broker, checkInterval time.Duration) (*Manager, error) {
```

Comment:
In the future, it may be a better plan to work more linearly towards the goal, so that we don't need these intermediate hacks to get us there.

That is, we could guarantee the creation of the MRC in a prior PR to putting this in

Reply:
Resolved offline. Marking as resolved